src/vm/codeversion.cpp

```diff
@@ -56,7 +56,12 @@ NativeCodeVersionNode::NativeCodeVersionNode(NativeCodeVersionId id, MethodDesc*
     m_parentId(parentId),
     m_pNextMethodDescSibling(NULL),
     m_id(id),
-    m_optTier(NativeCodeVersion::OptimizationTier0),
+#ifdef FEATURE_TIERED_COMPILATION
+    m_optTier(
+        pMethodDesc->RequestedAggressiveOptimization()
```

Comment:
Should this rather be initialized based on what was actually compiled?

Reply:
The tier is only applicable to methods eligible for tiering, that currently includes methods with r2r-pregenerated code and excludes methods with fragile ngen-pregenerated code. For the eligible methods, the only reason currently they would use tier 1 initially is if they are flagged with `AggressiveOptimization`. Whether the method has pregenerated code available is checked later before compiling the method.

Reply:
I meant whether this should be set based on what was JITed or going to be JITed.

If I am reading the code correctly, we will create a new NativeCodeVersionNode when we promote thing from Tier0 to Tier1; or when we actually JIT new code. We know the tier in both places.

The change is calling `RequestedAggressiveOptimization` on more places than what I would expect it to be called. So I am trying to understand why all the calls are really required.

Reply:
Oh ok, yes that should be avoidable, I'll take a look